### PR TITLE
EN-1584 SecretKeyAccessor - Multiple keys object

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
- Add ability to pass multiple keys with KeysObject
- Add KeysObject validation via io-ts lib (https://github.com/gcanti/io-ts)
- Add some jsdocs

Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [x] https://incountry.atlassian.net/browse/EN-1584

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [ ] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: in-crypt.js [hunk 1]
```diff
@@ -12,25 +12,43 @@ const AUTH_TAG_SIZE = 16;
...
+  * @param {import('./secret-key-accessor')} secretKeyAccessor
+  */
   constructor(secretKeyAccessor) {
     this._secretKeyAccessor = secretKeyAccessor;
   }
 
   async encryptAsync(text) {
-    const secret = await this._secretKeyAccessor.secureAccessor();
+    const keysObject = await this._secretKeyAccessor.secureAccessor();
+    const keyObj = keysObject.keys.find(k => k.keyVersion == keysObject.currentKeyVersion);
```
Comment (vladimir-grezin) [inline in-crypt.js]: Don't we need to check `keyObj` just like we do in `decryptAsync`?

File: secret-key-accessor.js [hunk 2]
```diff
@@ -1,29 +1,74 @@
...
+   * @return {KeysObject}
+   */
+  _wrapToKeysObject(key) {
+    return {
+      currentKeyVersion: 0,
+      keys: [{
+        key, 
+        keyVersion: 0
+      }]
+    }
```
Comment (RommeroInCountry) [inline secret-key-accessor.js]: @igor-alekseev-dev could you please make 0 a DEFAULT_VERSION constant? (and use it further in tests)

File: in-crypt.js (hunk 1 above)
Comment (avdyakovs) [inline in-crypt.js]: Maybe we should hide this logic inside secretKeyAccessor by introducing `currentSecureAccessor()` method? I guess there will be not only one place where we need to get latest key

Comment (igor-alekseev-dev) [reply in-crypt.js]: We have that check during the validation 
https://github.com/incountry/sdk-nodejs/pull/21/files#diff-6090367b8bcfdd046c9a2fee95dfd070R27

But yes to be 100% typesafe we need to check here also

File: package.json [hunk 3]
```diff
@@ -7,10 +7,12 @@
   "bugs": {
     "url": "https://github.com/incountry/sdk-nodejs/issues"
   },
-  "bundleDependencies": false,
+  "bundleDependencies": [],
   "dependencies": {
     "axios": "0.19.0",
     "dotenv": "8.0.0",
+    "fp-ts": "2.1.2",
```
Comment (RommeroInCountry) [inline package.json]: why we need this lib in deps?

File: tests/unit/crypt-key-accessor-tests.js [hunk 4]
```diff
@@ -7,25 +7,224 @@ const { expect } = chai;
...
+      expect(keyObj00.key).to.equal(secret0);
+
+      const secretKeyAccessor10 = new SecretKeyAccessor(async () => ({ keys: [{ keyVersion: 0, key: secret0 }, { keyVersion: 1, key: secret1 }, ], currentKeyVersion: 0 }));
+      const keyObj10 = await secretKeyAccessor10.getKey();
+      expect(keyObj10.key).to.equal(secret0);
+
+      const secretKeyAccessor20 = new SecretKeyAccessor(async () => ({ keys: [{ keyVersion: 0, key: secret0 }, { keyVersion: 1, key: secret1 }, ], currentKeyVersion: 1 }));
+      const keyObj20 = await secretKeyAccessor20.getKey();
+      expect(keyObj20.key).to.equal(secret1);
+    });
```
Comment (RommeroInCountry) [inline tests/unit/crypt-key-accessor-tests.js]: We can use parametrised tests for this instead (take a look at sdk-python or proxy integration tests). No need to do it now, just a head up

File: in-crypt.js [hunk 5]
```diff
@@ -12,25 +12,38 @@ const AUTH_TAG_SIZE = 16;
...
   }
 
-  async decryptAsync(s) {
+  /**
+   * 
+   * @param {string} s 
+   * @param {number} secretVersion 
+   */
+  async decryptAsync(s, secretVersion) {
+    const secret = await this._secretKeyAccessor.getSecret(secretVersion);
```
Comment (vladimir-grezin) [inline in-crypt.js]: Maybe we better name it somehow else, to avoid `secret.secret` things?

Comment (RommeroInCountry) [reply in-crypt.js]: We can actually use destructuring here

File: tests/unit/crypt-key-accessor-tests.js [hunk 6]
```diff
@@ -7,25 +7,224 @@ const { expect } = chai;
...
 describe('SecretKeyAccessor', () => {
   context('with asynchronous callback', () => {
-    it('should access secret key', async () => {
+    it('should access secret key if string has passed', async () => {
       const secret = 'supersecret';
       const secretKeyAccessor = new SecretKeyAccessor(async () => secret);
-      const key = await secretKeyAccessor.secureAccessor();
-      expect(key).to.equal(secret);
+      const secretObj = await secretKeyAccessor.getSecret();
+      expect(secretObj.secret).to.equal(secret);
```
Comment (RommeroInCountry) [inline tests/unit/crypt-key-accessor-tests.js]: I'd also check that version === SecretKeyAccessor.DEFAULT_VERSION

**Pull request merged**